File: src/Starfield2026.MiniToolbox/src/MiniToolbox.Core/Bntx/BinaryData/BinaryMemberAttribute.cs
using System;

namespace MiniToolbox.Core.Bntx;

[AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
public class BinaryMemberAttribute : Attribute
{
	internal static readonly BinaryMemberAttribute Default = new BinaryMemberAttribute();

	public int Offset { get; set; }

	public OffsetOrigin OffsetOrigin { get; set; }

	public BinaryBooleanFormat BooleanFormat { get; set; }

	public BinaryDateTimeFormat DateTimeFormat { get; set; }

	public BinaryStringFormat StringFormat { get; set; }

	public int Length { get; set; }

	public bool Strict { get; set; }

	public Type Converter { get; set; }
}

File: src/Starfield2026.MiniToolbox/src/MiniToolbox.Api/src/bntx/BinaryData/BinaryMemberAttribute.ts
import { BinaryBooleanFormat } from './BinaryBooleanFormat.js';
import { BinaryDateTimeFormat } from './BinaryDateTimeFormat.js';
import { BinaryStringFormat } from './BinaryStringFormat.js';
import { OffsetOrigin } from './OffsetOrigin.js';

/**
 * 1:1 port of BinaryMemberAttribute.cs
 * In C# this is a [BinaryMember] attribute for decorating fields/properties.
 * In TS, we represent it as a plain config object.
 */
export class BinaryMemberAttribute {
    static readonly Default = new BinaryMemberAttribute();

    offset: number = 0;
    offsetOrigin: OffsetOrigin = OffsetOrigin.Current;
    booleanFormat: BinaryBooleanFormat = BinaryBooleanFormat.NonZeroByte;
    dateTimeFormat: BinaryDateTimeFormat = BinaryDateTimeFormat.NetTicks;
    stringFormat: BinaryStringFormat = BinaryStringFormat.VariableLengthPrefix;
    length: number = 0;
    strict: boolean = false;
    converter: (new () => any) | null = null;
}
